gdal/swig/python/scripts/gdal2tiles.py

```diff
@@ -651,8 +651,15 @@ def __init__(self, arguments ):
         # KML generation
         self.kml = self.options.kml
 
-        # Output the results
+        # LC_CTYPE check
+        if 'UTF-8' not in os.environ.get("LC_CTYPE", ""):
```

Comment:
Could we emit the warning only if the filename is not ASCII ? because otherwise on Windows the warning will be always emitted since LC_CTYPE will not be defined there